modules/user_data/templates/tfe_ec2.sh.tpl

```diff
@@ -139,9 +139,51 @@ install_tfe() {
   arguments+=("disable-replicated-ui")
   %{ endif ~}
 
-  curl -o /tmp/install.sh https://get.replicated.com/docker/terraformenterprise/active-active
-  chmod +x /tmp/install.sh
-  /tmp/install.sh "$${arguments[@]}" | tee -a /var/log/ptfe.log
+  %{ if airgap_url != null ~}
+  arguments+=("airgap")
+  echo "[Terraform Enterprise] Installing Docker Engine from Repository" | tee -a $log_pathname
+if [[ $distribution == "ubuntu" ]]
+then
+  apt-get --assume-yes update
+  apt-get --assume-yes install \
+    ca-certificates \
+    curl \
+    gnupg \
+    lsb-release
+  curl --fail --silent --show-error --location https://download.docker.com/linux/ubuntu/gpg \
+    | gpg --dearmor --output /usr/share/keyrings/docker-archive-keyring.gpg
+  echo \
+    "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] \
+    https://download.docker.com/linux/ubuntu $(lsb_release --codename --short) stable" \
+    | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+  apt-get --assume-yes update
+  apt-get --assume-yes install docker-ce docker-ce-cli containerd.io
+  apt-get --assume-yes autoremove
+elif [[ $distribution == "rhel" ]]
+then
+  yum install --assumeyes yum-utils
+  yum-config-manager --add-repo https://download.docker.com/linux/rhel/docker-ce.repo
+  yum install --assumeyes docker-ce docker-ce-cli containerd.io
+fi
+replicated_filename="replicated.tar.gz"
+replicated_url="https://s3.amazonaws.com/replicated-airgap-work/$replicated_filename"
+replicated_pathname="$replicated_directory/$replicated_filename"
+echo "[Terraform Enterprise] Downloading Replicated from '$replicated_url' to '$replicated_pathname'" | tee -a $log_pathname
+curl --create-dirs --output "$replicated_pathname" "$replicated_url"
+echo "[Terraform Enterprise] Extracting Replicated in '$replicated_directory'" | tee -a $log_pathname
+tar --directory "$replicated_directory" --extract --file "$replicated_pathname"
+
+echo "[Terraform Enterprise] Copying airgap package '${airgap_url}' to '${airgap_pathname}'" | tee -a $log_pathname
+curl --create-dirs --output "${airgap_pathname}" "${airgap_url}"
+%{ else ~}
+curl -o /tmp/install.sh https://get.replicated.com/docker/terraformenterprise/active-active
+chmod +x /tmp/install.sh
+%{ endif ~}
+
+chmod +x $install_pathname
+cd $replicated_directory
+$install_pathname "$${arguments[@]}" | tee -a $log_pathname
+
```

Comment:
This content should be indented as it exists within the `install_tfe` function.

Reply:
Looks like this still needs to be indented, too.